core/src/main/java/org/infinispan/topology/CacheTopologyControlCommand.java

```diff
@@ -36,8 +36,8 @@
       JOIN,
       // A member is signaling that it wants to leave the cluster.
       LEAVE,
-      // A member is confirming that it finished the rebalance operation.
-      REBALANCE_CONFIRM,
+      // A member is confirming that it has finished the topology change during rebalance.
+      REBALANCE_PHASE_CONFIRM,
```

Comment:
The confirmation is sent after each phase, not just after state transfer. You have defined the terms in this way:

> state transfer is the actual transfer of the data, and the rebalance is the whole process of adding
a "pending" consistent hash, starting the state transfer, confirming the transfer of state, and finally replacing the "current" consistent hash with the pending one.

So we're confirming one of the phases of rebalance, not just state transfer.